### ISSUE TITLE
feature maps weighted sum by gumbel softmax

### ISSUE THREAD

**Issue by aragakiyuiii** [D-X-Y/AutoDL-Projects#61]
Hello, thanks for your great work! 
I want to ask you a question: In `lib/models/cell_searchs/search_cells.py, line58`, the weighted-sum in Gumbel-softmax is calculated as follows: `weigsum = sum( weights[_ie] * edge(nodes[j]) if _ie == argmaxs else weights[_ie] for _ie, edge in enumerate(self.edges[node_str]) )`. `weights[_ie] * edge(nodes[j])` is tensor but `weights[_ie]` is scalar. `weights[_ie]` does not multiply with a tensor like `edge(nodes[j])`. So `weights[_ie] * edge(nodes[j])` and `weights[_ie]` do not have the same dimension. 

However, code`weigsum = sum( weights[_ie] * edge(nodes[j]) if _ie == argmaxs else weights[_ie] for _ie, edge in enumerate(self.edges[node_str]) )` is all right when running. Why `weights[_ie] * edge(nodes[j])` and `weights[_ie]` can be added together even if both dimensions(data.shape) are different ?

Looking forward to your reply thanks.

**Comment by D-X-Y** [comment]
First of all, thanks for using my repo.
In PyTorch, the operations on tensor can support broadcast for */+, if you add/multiply a scalar with a tensor, it is ok.
